stdlib/http/src/main/java/org/ballerinalang/net/http/websocket/client/InitEndpoint.java

```diff
@@ -149,6 +138,104 @@ private static void populateClientConnectorConfig(MapValue<String, Object> clien
+     */
+    private static void populateRetryConnectorConfig(MapValue<String, Object> retryConfig,
+                                                     RetryContext retryConnectorConfig) {
+        setInterval(Integer.parseInt(retryConfig.get(INTERVAL).toString()), retryConnectorConfig);
+        setBackOfFactor(Float.parseFloat(retryConfig.get(BACK_OF_FACTOR).toString()),
+                retryConnectorConfig);
+        setMaxInterval(Integer.parseInt(retryConfig.get(MAX_INTERVAL).toString()),
+                retryConnectorConfig);
+        setMaxAttempts(Integer.parseInt(retryConfig.get(MAX_COUNT).toString()),
+                retryConnectorConfig);
+    }
+
+    /**
+     * Validate and create the webSocket service.
+     *
+     * @param clientEndpointConfig a client endpoint config
+     * @param strand a strand
+     * @return webSocketService
+     */
+    private static WebSocketService validateAndCreateWebSocketService(MapValue<String, Object> clientEndpointConfig,
+                                                                      Strand strand) {
+        Object clientService = clientEndpointConfig.get(WebSocketConstants.CLIENT_SERVICE_CONFIG);
+        if (clientService != null) {
+            BType param = ((ObjectValue) clientService).getType().getAttachedFunctions()[0].getParameterType()[0];
+            if (param == null || !WebSocketConstants.WEBSOCKET_CLIENT_NAME.equals(
+                    param.toString())) {
+                throw new WebSocketException("The callback service should be a WebSocket Client Service");
+            }
+            return new WebSocketService((ObjectValue) clientService, strand.scheduler);
+        } else {
+            return new WebSocketService(strand.scheduler);
+        }
+    }
+
+    /**
+     * Set value of the interval in the retry config.
+     *
+     * @param interval retry interval
+     * @param retryConnectorConfig retry connector config
+     */
+    private static void setInterval(int interval, RetryContext retryConnectorConfig) {
+        if (interval < 0) {
+            logger.warn("{} The interval[{}] value is set to 1000.", STATEMENT_FOR_INTEVAL, interval);
+            interval = 1000;
+        }
+        retryConnectorConfig.setInterval(interval);
+    }
+
+    /**
+     * Set value of the backOfFactor in the retry config.
+     *
+     * @param backOfFactor a rate of increase of the reconnect delay
+     * @param retryConnectorConfig retry connector config
+     */
+    private static void setBackOfFactor(float backOfFactor, RetryContext retryConnectorConfig) {
+        if (backOfFactor < 0) {
+            logger.warn("{} The backOfFactor[{}] value is set to 1.0", STATEMENT_FOR_BACK_OF_FACTOR,
+                    backOfFactor);
+            backOfFactor = (float) 1.0;
+        }
+        retryConnectorConfig.setBackOfFactor(backOfFactor);
+    }
+
+    /**
+     * Set value of the maxInterval in the retry config.
+     *
+     * @param maxInterval a maximum interval
```

Comment:
```suggestion
     * @param maxInterval - the maximum interval.
```
Can we explain this a bit more in the description?